Source/WebKit/NetworkProcess/curl/NetworkDataTaskCurl.cpp

```diff
@@ -61,17 +61,18 @@ NetworkDataTaskCurl::NetworkDataTaskCurl(NetworkSession& session, NetworkDataTas
     , m_sourceOrigin(parameters.sourceOrigin)
 {
     auto request = parameters.request;
-    if (request.url().protocolIsInHTTPFamily()) {
-        if (m_storedCredentialsPolicy == StoredCredentialsPolicy::Use) {
-            auto url = request.url();
-            m_user = url.user();
-            m_password = url.password();
-            request.removeCredentials();
-
+    auto url = request.url();
+    if (m_storedCredentialsPolicy == StoredCredentialsPolicy::Use && url.protocolIsInHTTPFamily()) {
+        m_user = url.user();
+        m_password = url.password();
+        request.removeCredentials();
+        url = request.url();
+
+        if (auto* storageSession = m_session->networkStorageSession()) {
```

Comment:
As per the new guildline, we shouldn't call methods without keeping a ref pointer.
https://github.com/WebKit/WebKit/wiki/Smart-Pointer-Usage-Guidelines
> RefPtr storageSession = m_session->networkStorageSession()

Reply:
NetworkSession::networkStorageSession() has nothing to do with RefPtr, but does it still need to be RefPtr?

https://github.com/WebKit/WebKit/blob/74a570d11375304d6dd2fe7e2e2696e5f93bcff2/Source/WebKit/NetworkProcess/NetworkSession.cpp#L90-L97

https://github.com/WebKit/WebKit/blob/74a570d11375304d6dd2fe7e2e2696e5f93bcff2/Source/WebKit/NetworkProcess/NetworkProcess.cpp#L563-L566

https://github.com/WebKit/WebKit/blob/74a570d11375304d6dd2fe7e2e2696e5f93bcff2/Source/WebKit/NetworkProcess/NetworkProcess.h#L549

Reply:
You are right. RefPtr isn't needed here.